pkg/controller/garbagecollector/patch.go

```diff
@@ -52,3 +52,8 @@ func (n *node) patchToUnblockOwnerReferences() ([]byte, error) {
 	dummy.ObjectMeta.UID = n.identity.UID
 	return json.Marshal(dummy)
 }
+
+func deleteFinalizerPatch(finalizer string, UID types.UID) []byte {
+	patch := fmt.Sprintf(`{"metadata":{"$deleteFromPrimitiveList/finalizers":["%s"],"uid":"%s"}}`, finalizer, UID)
```

Comment:
form this by json-encoding a structured object... we don't want to worry about safety of the finalizer string here

Reply:
Are you concerned about illegal patch format or illegal `finalizer string`?

Reply:
both, though now I see this is called with fixed finalizer values. building json this way with object input is unsettling, but I can live with it

Reply:
Regarding the patch format, I shared the same concern, but I did't build the patch via `CreateTwoWayMergePatch` because
* I'll have to make a fake addition to keep the "uid" precondition.
* The generated patch contains `"$setElementOrder/finalizers":<remaining finalizers>`, which will increase the chance of collision.

Besides, the unit test makes sure the patch is valid. Of course it doesn't protect us against version skew, but neither does `CreateTwoWayMergePatch`.

Regarding the finalizer string, too bad we didn't define a `type Finalizer string`. I think we can still make that change?